### ISSUE TITLE
Video does not open when high quality is restricted.

### ISSUE THREAD

**Issue by BenjaminSF** [lindstrm/twitcha#19]
Some Twitch-channels restrict the best video quality to subscribers only. E.g. "source"-quality is only available for subscribers on the channel sc2proleague. It's a fairly niche practice, I've only seen it on korean esports channels, but currently it doesn't work when the default quality is also set to Source.

**Comment by yukihyou** [comment]
Just going to tack onto this issue with regards to the changes that have been put in recently to support the newer Twitch channel formats. There are a number of channels that use strange values (see below), however there is always a "best" option, even if source/high/low/720p/etc doesn't work.

```[cli][info] Found matching plugin twitch for URL twitch.tv/thevehementrose
[plugin.twitch][info] Attempting to authenticate using OAuth token
[plugin.twitch][info] Successfully logged in as YukiHyou
Available streams: 614p (worst, best), audio_only```

What happens currently is that Twitcha will continuously attempt to start Livestreamer, with different quality arguments, and cycle around through them all over and over, with the channel highlighted green. This process cannot be stopped without closing Twitcha or toggling the "Use Livestreamer" option to stop the cycle.

Could I suggest either:
- Parsing the last line returned by Livestreamer after the first failure to update the right-click list with the correct values available for that stream if the defaults aren't sufficient; and/or
- Allow the Default Quality box to be freetext instead of a dropdown, so I can put `best` in there and work around this problem.

Currently the workaround is to manually start livestreamer with `best` quality, however that somewhat defeats the purpose of Twitcha.

**Comment by lindstrm** [comment]
Are you sure the "best" option is available in Livestreamer, I've only seen it with streamlink?

Anyway this issue with abnormal resolutions has been fixed in 2.1.15, thanks for your comment though.
